pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -516,10 +517,15 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings,
 				},
 			}
 		} else {
-			cluster.TlsContext.CommonTlsContext.ValidationContextType = model.ConstructValidationContext(model.CARootCertPath)
+			cluster.TlsContext.CommonTlsContext.ValidationContextType = model.ConstructValidationContext(model.CARootCertPath, tls.SubjectAltNames)
 			cluster.TlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{}
 			refreshDuration, _ := ptypes.Duration(meshConfig.SdsRefreshDelay)
 			for _, sa := range serviceAccounts {
+				// Skip service account which is added from annotation 'alpha.istio.io/canonical-serviceaccounts'
```

Comment:
I think a better comment would be 'why we want to something' instead of 'what it does'  :)

Reply:
+1, other than that, the PR LGTM.

Reply:
added more comments to explain the reason.